pkg/ccl/sqlproxyccl/forwarder.go

```diff
@@ -524,6 +539,9 @@ func (p *processor) waitResumed(ctx context.Context) error {
 		if ctx.Err() != nil {
 			return ctx.Err()
 		}
+		if p.mu.closed {
```

Comment:
nit: I think we should remove the ctx argument from waitResumed. waitResumed does not really respect context cancellation. the only way to get out of the method is for the method to succeed or to call .close() on the processor.

Reply:
Discussed offline. Will leave this as-is until we revisit the use of condition variables again.